### PR TITLE
Enable replying from the push notification

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Add the ability to reply to a push notification without opening the app.

Note: On Android you need Android 7.0+ for lower version it will not show the option to reply

#### Device Information
This PR was tested on: 
* IOS 10.3 iPhone 6s
* Android 6.0.1 does not give the option to reply
* Android 7.0 and Android 7.1 Simulators

### CODE REVIEW

File: android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java [hunk 1]
```diff
@@ -173,19 +179,49 @@ protected void postNotification(int id, Notification notification) {
...
         notification.setDeleteIntent(deleteIntent);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
             notification.setGroup(GROUP_KEY_MESSAGES);
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            try {
+                //Use reflection to remove all old actions
+                Field f = notification.getClass().getDeclaredField("mActions");
```
Comment (hmhealey) [inline android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java]: What is this doing? Using reflection like this seems ridiculously hacky

Comment (enahum) [reply android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java]: so there is no way to access `mActions`without reflection for the NotificationBuilder and we need that in order to clear it and set the "updated" action

Comment (enahum) [reply android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java]: This action https://github.com/mattermost/mattermost-mobile/pull/871/files#diff-d29c058681ba74acb23997331d3d778eR222.

But I found a way around this and a bug that I need to fix

**Comment by enahum** [comment]
@hmhealey this should be ready now

File: android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java [hunk 2]
```diff
@@ -173,19 +179,40 @@ protected void postNotification(int id, Notification notification) {
...
         delIntent.putExtra(NOTIFICATION_ID, notificationId);
-        PendingIntent deleteIntent = PendingIntent.getBroadcast(mContext, 0, delIntent, 0);
+        PendingIntent deleteIntent = NotificationIntentAdapter.createPendingNotificationIntent(mContext, delIntent, mNotificationProps);
         notification.setDeleteIntent(deleteIntent);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
             notification.setGroup(GROUP_KEY_MESSAGES);
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```
Comment (enahum) [inline android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java]: Wondering if I should add in the condition that postId es not null or do I leave it like this, leaving it as it is will let the user post to the same channel but not as a reply.

Thoughts?

File: ios/MattermostManaged.m [hunk 3]
```diff
@@ -74,6 +74,13 @@ - (void) remoteConfigChanged {
   exit(0);
 }
 
+RCT_EXPORT_METHOD(quitAppWithDelayInSeconds:(double)delay) {
```
Comment (enahum) [inline ios/MattermostManaged.m]: Note to self remove this code as is not needed

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
     "react-intl": "2.3.0",
     "react-native": "0.44.0",
     "react-native-animatable": "1.2.2",
+    "react-native-background-timer": "1.2.0",
```
Comment (enahum) [inline package.json]: Also remove this

**Pull request merged**